### PR TITLE
.github/workflows/create-backport-trackers.yml: update actions

### PR DESCRIPTION
**Pull request by dmick**
Getting warning about node16 being deprecated.  The workflow doesn't use node directly, but through the external actions.  Moving to node20 requires changing setup-python version; Bhacaz/checkout-files is deprecated and recommends actions/checkout.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by dmick** [comment]
Tested by manually running workflow on a modified workflow that removed the "if : github.ref == 'refs/heads/main'"

File: .github/workflows/create-backport-trackers.yml [hunk 1]
```diff
@@ -36,10 +36,11 @@ jobs:
     runs-on: ubuntu-latest
     if: github.ref == 'refs/heads/main'
     steps:
-      - uses: Bhacaz/checkout-files@e3e34e7daef91a5f237485bb88a260aee4be29dd
+      - uses: actions/checkout@v4
         with:
-          files: src/script/backport-create-issue src/script/requirements.backport-create-issue.txt
-      - uses: actions/setup-python@v4
+          sparse-checkout: src/script/backport-create-issue src/script/requirements.backport-create-issue.txt
```
Comment (phlogistonjohn) [inline .github/workflows/create-backport-trackers.yml:41]: If it works, great! However, I see in the docs:
```
    # Do a sparse checkout on given patterns. Each pattern should be separated with
    # new lines.
    # Default: null
    sparse-checkout: ''
```

So I don't know if we should also change the line to something like:
```suggestion
          sparse-checkout: |
            src/script/backport-create-issue
            src/script/requirements.backport-create-issue.txt
```

**Review by phlogistonjohn — COMMENTED** [review]
If you are confident in your test and don't want to bother with my suggestion let me know and I can approve anyway. Otherwise, even if space separation woks we can change it to match the docs and possibly future proof our config.

File: .github/workflows/create-backport-trackers.yml (hunk 1 above)
Comment (dmick) [reply .github/workflows/create-backport-trackers.yml:41]: I can change that.  I don't think it matters to the yaml, but it does parse slightly differently (in both cases it's a list, but in the | case each string has an appended newline).

**Comment by dmick** [comment]
Thanks @ktdreyer for providing the copr.

**Pull request merged**